version: "3.8"

services:
  db:
    container_name: control_db
    image: postgres:14-bullseye
    restart: on-failure
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    env_file:
      - ./.env
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready" ]
      interval: 10s
      timeout: 3s
      retries: 3
    networks:
      - control
    ports:
      - "5433:5432"

  nginx:
    container_name: control_nginx
    image: nginx:stable-alpine
    restart: on-failure
    ports:
      - "4000:4000"
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf
    depends_on:
      - app
      - frontend
    networks:
      - control

  app:
    container_name: control_app
    image: akafer/ins_control
    restart: on-failure
    ports:
      - "8500:8500"
    command: >
      bash -c "alembic upgrade head
      && python3 /app/src/scripts/create_user.py
      && python3 /app/src/run.py --port=8500"
    env_file:
      - .env
    volumes:
        - ./files/instructions:/app/src/static/instructions
        - ./files/signatures:/app/src/static/signatures
        - ./files/templates:/app/src/static/templates
        - ./files/training_modules:/app/src/static/training_modules
        - ./logs:/app/src/logs
    depends_on:
      db:
        condition: service_healthy
    networks:
      - control

  frontend:
    container_name: control_frontend
    image: akafer/ins_control_frontend
    restart: on-failure
    env_file:
      - .env
    ports:
      - "3000:3000"
    depends_on:
      - app
    networks:
      - control

volumes:
  postgres_data:

networks:
  control:
    name: control
    driver: bridge
